src/build_system.ml

```diff
@@ -183,6 +131,55 @@ module Internal_rule = struct
        [if_file_exists] are resolved inside the [Build.t] value. *)
     ignore (Lazy.force t.static_deps : Build_interpret.Static_deps.t);
     Build_interpret.lib_deps t.build
+
+  let root =
+    { id          = Id.gen ()
+    ; static_deps = lazy { rule_deps = Path.Set.empty
+                         ; action_deps = Path.Set.empty
+                         }
+    ; targets     = Path.Set.empty
+    ; context     = None
+    ; build       = Build.return (Action.Progn [])
+    ; mode        = Standard
+    ; loc         = None
+    ; dir         = Path.root
+    ; exec        = Not_started { eval_rule = (fun _ -> assert false)
```

Comment:
Is there any way to encode this into a new constructor so that these assertions can be removed?

Reply:
I'm not sure how to interpret it though. This rule represent the build goal, so it should only terminate when the build is done. We could also create a new rule for every build and make it depend on the requested target. I'm having a look

Reply:
This is a bit annoying to change actually, so I just added a comment for now. We can look at this in another PR

Reply:
OK! that's good like this.